core/client/src/main/java/alluxio/client/file/UnderFileSystemFileInStream.java

```diff
@@ -164,7 +164,10 @@ private ByteBuffer allocateBuffer() {
   private void bufferedRead(int len) throws IOException {
     mBuffer.clear();
     int bytesRead = directRead(mBuffer.array(), 0, len);
```

Comment:
this is tricky.  according to https://docs.oracle.com/javase/7/docs/api/java/nio/Buffer.html#flip(), `Flips this buffer. The limit is set to the current position and then the position is set to zero. If the mark is defined then it is discarded.` What is the implication on the underlying array size? do we know for sure we have `len` bytes?

It seems fine as the only reference of `bufferedRead` is `bufferedRead(mBuffer.capacity());` which meets this assumption. However, this assumption is not documented or enforced anywhere and error prone in the future. I would suggest to change `bufferRead` to take no parameter, and replace `len` with `mBuffer.capacity()` inside `bufferedRead`